File: scenarios/basic/mobile/android-views/InaccessibleTooltip.java
package com.example.accessibilityscenarios;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.TextView;

public class InaccessibleTooltipActivity extends Activity {
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        ScrollView scrollView = new ScrollView(this);
        LinearLayout mainLayout = new LinearLayout(this);
        mainLayout.setOrientation(LinearLayout.VERTICAL);
        mainLayout.setPadding(32, 32, 32, 32);
        mainLayout.setBackgroundColor(0xFFF5F5F5);
        
        LinearLayout headerLayout = createHeader();
        mainLayout.addView(headerLayout);
        
        LinearLayout formLayout = createForm();
        mainLayout.addView(formLayout);
        
        scrollView.addView(mainLayout);
        setContentView(scrollView);
    }
    
    private LinearLayout createHeader() {
        LinearLayout headerLayout = new LinearLayout(this);
        headerLayout.setOrientation(LinearLayout.VERTICAL);
        headerLayout.setBackgroundColor(0xFF2196F3);
        headerLayout.setPadding(32, 32, 32, 32);
        
        TextView titleText = new TextView(this);
        titleText.setText("Account Registration");
        titleText.setTextSize(24);
        titleText.setTextColor(0xFFFFFFFF);
        titleText.setTypeface(null, android.graphics.Typeface.BOLD);
        
        TextView subtitleText = new TextView(this);
        subtitleText.setText("Create your account to get started");
        subtitleText.setTextSize(16);
        subtitleText.setTextColor(0xE6FFFFFF);
        
        headerLayout.addView(titleText);
        headerLayout.addView(subtitleText);
        
        return headerLayout;
    }
    
    private LinearLayout createForm() {
        LinearLayout formLayout = new LinearLayout(this);
        formLayout.setOrientation(LinearLayout.VERTICAL);
        formLayout.setBackgroundColor(0xFFFFFFFF);
        formLayout.setPadding(20, 20, 20, 20);
        
        TextView formTitle = new TextView(this);
        formTitle.setText("Personal Information");
        formTitle.setTextSize(20);
        formTitle.setTypeface(null, android.graphics.Typeface.BOLD);
        formTitle.setTextColor(0xFF1976D2);
        formTitle.setPadding(0, 0, 0, 20);
        
        formLayout.addView(formTitle);
        
        String[][] fields = {
            {"Email Address", "Enter your email", "We'll use this to send you important updates and notifications about your account."},
            {"Password", "Enter your password", "Password must be at least 8 characters long and contain uppercase, lowercase, and numbers."},
            {"Phone Number", "Enter your phone number", "Include country code. Example: +1 (555) 123-4567"},
            {"Date of Birth", "MM/DD/YYYY", "You must be at least 18 years old to create an account."},
            {"Security Question", "Select a security question", "Choose a question that only you know the answer to. This helps secure your account."}
        };
        
        for (String[] field : fields) {
            LinearLayout fieldLayout = createFormField(field[0], field[1], field[2]);
            formLayout.addView(fieldLayout);
        }
        
        Button submitButton = new Button(this);
        submitButton.setText("Create Account");
        submitButton.setBackgroundColor(0xFF2196F3);
        submitButton.setTextColor(0xFFFFFFFF);
        submitButton.setPadding(16, 16, 16, 16);
        
        LinearLayout.LayoutParams submitParams = new LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
        submitParams.setMargins(0, 20, 0, 0);
        formLayout.addView(submitButton, submitParams);
        
        return formLayout;
    }
    
    private LinearLayout createFormField(String label, String placeholder, String helpText) {
        LinearLayout fieldLayout = new LinearLayout(this);
        fieldLayout.setOrientation(LinearLayout.VERTICAL);
        fieldLayout.setPadding(0, 0, 0, 16);
        
        // Field Label and Help Icon
        LinearLayout labelLayout = new LinearLayout(this);
        labelLayout.setOrientation(LinearLayout.HORIZONTAL);
        labelLayout.setPadding(0, 0, 0, 8);
        
        TextView labelText = new TextView(this);
        labelText.setText(label);
        labelText.setTextSize(16);
        labelText.setTypeface(null, android.graphics.Typeface.BOLD);
        labelText.setTextColor(0xFF1976D2);
        
        // Help Icon - INACCESSIBLE TOOLTIP
        ImageButton helpButton = new ImageButton(this);
        helpButton.setImageResource(android.R.drawable.ic_menu_help);
        helpButton.setBackgroundColor(android.graphics.Color.TRANSPARENT);
        helpButton.setContentDescription(null); // MISSING: No accessible description
        helpButton.setPadding(8, 8, 8, 8);
        helpButton.setMinimumWidth(40);
        helpButton.setMinimumHeight(40);
        
        // Tooltip - Only shows on hover/click, not keyboard accessible
        final TextView tooltipText = new TextView(this);
        tooltipText.setText(helpText);
        tooltipText.setTextSize(12);
        tooltipText.setTextColor(0xFFFFFFFF);
        tooltipText.setBackgroundColor(0xFF424242);
        tooltipText.setPadding(8, 8, 8, 8);
        tooltipText.setVisibility(View.GONE);
        
        helpButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (tooltipText.getVisibility() == View.GONE) {
                    tooltipText.setVisibility(View.VISIBLE);
                } else {
                    tooltipText.setVisibility(View.GONE);
                }
            }
        });
        
        labelLayout.addView(labelText);
        labelLayout.addView(helpButton);
        
        // Input Field
        EditText inputField = new EditText(this);
        inputField.setHint(placeholder);
        inputField.setPadding(16, 16, 16, 16);
        inputField.setBackgroundColor(0xFFF0F0F0);
        
        fieldLayout.addView(labelLayout);
        fieldLayout.addView(inputField);
        fieldLayout.addView(tooltipText);
        
        return fieldLayout;
    }
}

File: scenarios/basic/mobile/android-compose/InaccessibleTooltip.kt
package com.example.accessibilityscenarios

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun InaccessibleTooltipScreen() {
    val formFields = listOf(
        FormField(
            id = 1,
            label = "Email Address",
            placeholder = "Enter your email",
            type = "email",
            helpText = "We'll use this to send you important updates and notifications about your account."
        ),
        FormField(
            id = 2,
            label = "Password",
            placeholder = "Enter your password",
            type = "password",
            helpText = "Password must be at least 8 characters long and contain uppercase, lowercase, and numbers."
        ),
        FormField(
            id = 3,
            label = "Phone Number",
            placeholder = "Enter your phone number",
            type = "tel",
            helpText = "Include country code. Example: +1 (555) 123-4567"
        ),
        FormField(
            id = 4,
            label = "Date of Birth",
            placeholder = "MM/DD/YYYY",
            type = "date",
            helpText = "You must be at least 18 years old to create an account."
        ),
        FormField(
            id = 5,
            label = "Security Question",
            placeholder = "Select a security question",
            type = "select",
            helpText = "Choose a question that only you know the answer to. This helps secure your account."
        )
    )
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF5F5F5))
            .padding(16.dp)
    ) {
        // Header
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFF2196F3))
        ) {
            Column(
                modifier = Modifier.padding(20.dp)
            ) {
                Text(
                    text = "Account Registration",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = "Create your account to get started",
                    fontSize = 16.sp,
                    color = Color.White.copy(alpha = 0.9f)
                )
            }
        }
        
        // Form
        Card(
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(
                modifier = Modifier.padding(20.dp)
            ) {
                Text(
                    text = "Personal Information",
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color(0xFF1976D2),
                    modifier = Modifier.padding(bottom = 20.dp)
                )
                
                LazyColumn(
                    verticalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    items(formFields) { field ->
                        FormFieldCard(field = field)
                    }
                }
                
                // Submit Button
                Button(
                    onClick = { },
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(top = 20.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF2196F3))
                ) {
                    Text(
                        text = "Create Account",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        }
    }
}

@Composable
fun FormFieldCard(field: FormField) {
    var showTooltip by remember { mutableStateOf(false) }
    
    Column(
        modifier = Modifier.fillMaxWidth()
    ) {
        // Field Label and Help Icon
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = field.label,
                fontSize = 16.sp,
                fontWeight = FontWeight.Medium,
                color = Color(0xFF1976D2)
            )
            
            Spacer(modifier = Modifier.width(8.dp))
            
            // Help Icon - INACCESSIBLE TOOLTIP
            Box {
                IconButton(
                    onClick = { showTooltip = !showTooltip },
                    modifier = Modifier.size(20.dp) // Small touch target
                ) {
                    Icon(
                        imageVector = Icons.Default.Help,
                        contentDescription = null, // MISSING: No accessible description
                        tint = Color.Gray,
                        modifier = Modifier.size(16.dp)
                    )
                }
                
                // Tooltip - Only shows on hover/click, not keyboard accessible
                if (showTooltip) {
                    Card(
                        modifier = Modifier
                            .padding(top = 4.dp)
                            .background(Color.Transparent),
                        colors = CardDefaults.cardColors(containerColor = Color(0xFF424242)),
                        elevation = CardDefaults.cardElevation(defaultElevation = 8.dp)
                    ) {
                        Text(
                            text = field.helpText,
                            fontSize = 12.sp,
                            color = Color.White,
                            modifier = Modifier.padding(8.dp),
                            maxLines = 3
                        )
                    }
                }
            }
        }
        
        // Input Field
        when (field.type) {
            "email", "tel", "date" -> {
                OutlinedTextField(
                    value = "",
                    onValueChange = { },
                    placeholder = { Text(field.placeholder) },
                    modifier = Modifier.fillMaxWidth(),
                    singleLine = true
                )
            }
            "password" -> {
                OutlinedTextField(
                    value = "",
                    onValueChange = { },
                    placeholder = { Text(field.placeholder) },
                    modifier = Modifier.fillMaxWidth(),
                    singleLine = true,
                    visualTransformation = PasswordVisualTransformation()
                )
            }
            "select" -> {
                var expanded by remember { mutableStateOf(false) }
                val options = listOf(
                    "What was your first pet's name?",
                    "What city were you born in?",
                    "What was your mother's maiden name?",
                    "What was the name of your first school?"
                )
                
                ExposedDropdownMenuBox(
                    expanded = expanded,
                    onExpandedChange = { expanded = !expanded }
                ) {
                    OutlinedTextField(
                        value = "",
                        onValueChange = { },
                        readOnly = true,
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded) },
                        modifier = Modifier
                            .fillMaxWidth()
                            .menuAnchor()
                    )
                    
                    ExposedDropdownMenu(
                        expanded = expanded,
                        onDismissRequest = { expanded = false }
                    ) {
                        options.forEach { option ->
                            DropdownMenuItem(
                                text = { Text(option) },
                                onClick = { expanded = false }
                            )
                        }
                    }
                }
            }
        }
    }
}

data class FormField(
    val id: Int,
    val label: String,
    val placeholder: String,
    val type: String,
    val helpText: String
)
